index.js

```diff
@@ -102,6 +110,59 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  if (!fastify.hasRequestDecorator('pg')) {
+    fastify.decorateRequest('pg', null)
+  }
+
+  fastify.addHook('onRoute', routeOptions => {
+    const transactionConnection = routeOptions.pg && routeOptions.pg.transact
+
+    if (transactionConnection) {
+      const preHandler = async (req, reply) => {
+        const client = await pool.connect()
+
+        if (name) {
+          req.pg = {}
```

Comment:
this works but it's not great to assign it an empty object the populate a property on line 132. just do this on line 132:

```js
req.pg = { [name]: client }
```

Reply:
actually no, ignore the above. you can't unconditionally set req.pg to an empty object, because another registration of the plugin may have already set something on it. you must conditionally initialize it if it's not yet initialized